src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.previous = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Incorrect index: " + index);
+        }
+        Node<T> newNode = new Node<>(value);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            newNode.next = head;
+            head.previous = newNode;
+            head = newNode;
+            size++;
+        } else {
+            Node<T> current = getNodeByIndex(index);
+            newNode.next = current;
+            newNode.previous = current.previous;
+            current.previous.next = newNode;
+            current.previous = newNode;
+            size++;
```

Comment:
it makes sense to extract the logic from `else if` and `else` to separate method `addFirst` and `addAt`